pyomo/contrib/interior_point/tests/test_reg.py

```diff
@@ -0,0 +1,151 @@
+import pyutilib.th as unittest
+from pyomo.common.dependencies import attempt_import
+
+np, numpy_available = attempt_import('numpy', 'Interior point requires numpy',
+        minimum_version='1.13.0')
+scipy, scipy_available = attempt_import('scipy', 'Interior point requires scipy')
+mumps_interface, mumps_available = attempt_import(
+        'pyomo.contrib.interior_point.linalg.mumps_interface',
+        'Interior point requires mumps')
+if not (numpy_available and scipy_available):
+    raise unittest.SkipTest('Interior point tests require numpy and scipy')
+
+from pyomo.contrib.pynumero.extensions.asl import AmplInterface
+asl_available = AmplInterface.available()
+
+from pyomo.contrib.interior_point.interior_point import InteriorPointSolver
+from pyomo.contrib.interior_point.interface import InteriorPointInterface
+from pyomo.contrib.interior_point.linalg.scipy_interface import ScipyInterface
+
+
+class TestRegularization(unittest.TestCase):
+    @unittest.skipIf(not asl_available, 'asl is not available')
+    @unittest.skipIf(not mumps_available, 'mumps is not available')
+    def test_regularize_mumps(self):
+        interface = InteriorPointInterface('reg.nl')
+        '''This NLP is the solve for consistent initial conditions 
+        in a simple 3-reaction CSTR.'''
+
+        linear_solver = mumps_interface.MumpsInterface()
+
+        ip_solver = InteriorPointSolver(linear_solver,
+                                        regularize_kkt=True)
+
+        interface.set_barrier_parameter(1e-1)
+
+        # Evaluate KKT matrix before any iterations
+        kkt = interface.evaluate_primal_dual_kkt_matrix()
+        with self.assertRaises(RuntimeError):
+            # Should be Mumps error: -10, numerically singular
+            # (Really the matrix is structurally singular, but it has 
+            # enough symbolic zeros that the symbolic factorization can
+            # be performed.
+            linear_solver.do_symbolic_factorization(kkt)
+            linear_solver.do_numeric_factorization(kkt)
+
+        # Perform one iteration of interior point algorithm
+        x, duals_eq, duals_ineq = ip_solver.solve(interface, max_iter=1)
+
+# The exact regularization coefficient at which Mumps recognizes the matrix
+# as non-singular appears to be non-deterministic...
+# I have seen 1e-4, 1e-2, and 1e0.
+# According to scipy, 1e-4 seems to be correct
+# MUMPS 5.3.1 seems to settle on 1e-2
+#        # Expected regularization coefficient:
+        self.assertAlmostEqual(ip_solver.reg_coef, 1e-2)
```

Comment:
@carldlaird This is the case where I got self-inconsistent behavior with MUMPS 5.2.1 and inconsistent-with-scipy behavior with 5.3.1